tests/libvmi/storage.go

```diff
@@ -28,8 +30,26 @@ import (
 func WithContainerImage(name string) Option {
 	return func(vmi *kvirtv1.VirtualMachineInstance) {
 		diskName := "disk0"
-		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, newDisk(diskName, v1.DiskBusVirtio))
-		vmi.Spec.Volumes = append(vmi.Spec.Volumes, newContainerVolume(diskName, name))
+		replaceDisk(vmi, newDisk(diskName, v1.DiskBusVirtio))
```

Comment:
If the disk is already there, why would you like to replace it? I guess I am asking why `addDisk` is not a fit.

Reply:
That way we have single and simple `libvmi.NewAlpine(` that uses containerImage, but in tests one can provide a PVC or DV with different content. The same for Cirros/Fedora. 

I suppose we could have a full blown framework with API to add/replace disks. But here I went for the simplest solution that works in current tests.

Reply:
I am not sure that what we had so far is good, but let me try to see if I get this right:
Until now, we had two options: Some build options added disk (without checking that another with the same name existed already) and some build options ignored cases where the disks or volumes have been there already.

Now, the suggestion is that all build options that handle storage will either add or replace/override what was there before.
This may be fine, but it may also cause confusion.

I can share what originally guided me, and you take it from there and see if it makes sense and works with the new additions:
- If we need to add a volume, we add it to a disk that may either be there or not. If it is there already, we do nothing. If it is not there, we add it. More or less a lazy creation of the dependency.
- There are disks for container images and for cloud-init at the moment. (looks like the container image one is not following this rule, which I guess is bad).

So if we want to override the disk or volume, we wither need to choose which pattern makes more sense.
I guess another option is to explicitly require to override everything (e.g. `WithCleanDisks()`) or as you suggested, to always override (last one wins). But what happens when we have multiple volumes on the same disk?

I think you are more qualified to answer this one, so I will follow with what you think it fits better. We just need to make sure that nothing got broken by this.

Reply:
All the options like Memory or Rng are written in a way that last one wins, so if we have a default, a user can always override it. This is also the reason why I think it is good to have the semantics of "last wins". In my opinion `WithXyz` should just set the value, and the last one should win. 

With ContainerImage before this PR it was strange because it just added a new disk with the same name. I am not sure if that is even valid. If we add only when the disk is not there then we just take the first value, which is valid but is not useful and not easy to use. I would need to add `NewAlpineWithoutDisk`. 

My only question is, what do you mean by " what happens when we have multiple volumes on the same disk"?

Reply:
Not sure if I am not totally mistaken, the hierarchy in the VMI spec is super confusing (me).
I thought that a disk can have multiple volumes.

I think the difference from the other two examples you gave is that here we have things that depend on each other.
Adding a volume depends on a disk to exist.

Anyway, I am fine with whatever you prefer to have. If this will fail in some future scenario, we should be able to fix it.

Reply:
I do not see any dependency here, 
I just add PVC, and I just add other alpine options, no dependencies at all.